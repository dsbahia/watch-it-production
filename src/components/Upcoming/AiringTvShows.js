import React, { useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faInfoCircle } from "@fortawesome/free-solid-svg-icons";
import MoreDetails from "../MoreDetails";
import Favourite from "../Favourite";
import YouTubeIcon from "../YouTubeIcon/YouTubeIcon";
import NoImagePlaceholder from "../../images/No-Image-Placeholder.png";
import "../../styles/topratedmovies.css";

function AiringTvShow({ posterpath, title, tvId, tvshowTrailer }) {
  const [isHovered, setIsHovered] = useState(false);
  const [isShown, setIsShown] = useState(false);
  const type = "tv";

  const handleClick = () => {
    setIsShown((current) => !current);
  };
  const posterCheck = () => {
    if (posterpath === null) {
      return NoImagePlaceholder;
    }
    return `https://image.tmdb.org/t/p/original/${posterpath}`;
  };
  return (
    <div className="top-rated">
      <div className="top-rated-poster">
        {" "}
        {tvshowTrailer ? (
          <a href={tvshowTrailer} target="_blank" rel="noopener noreferrer">
            <img
              className="top-rated-poster-img"
              alt={`${title} Tv Show poster`}
              src={posterCheck()}
              onMouseEnter={() => setIsHovered(true)}
              onMouseLeave={() => setIsHovered(false)}
            />
            {isHovered && <YouTubeIcon />}
          </a>
        ) : (
          <img
            className="top-rated-poster-img"
            alt={`${title} Tv Show poster`}
            src={posterCheck()}
          />
        )}
      </div>
      <div className="top-rated-title">{title}</div>
      <div className="more-details">
        <div className="buttons">
          <div className="favourites">
            <Favourite />
          </div>
          <button
            className="more-details-link"
            type="button"
            onClick={handleClick}
          >
            More Details
            <FontAwesomeIcon icon={faInfoCircle} className="icon" />
          </button>
        </div>
        {isShown ? <MoreDetails type={type} id={tvId} /> : null}
      </div>
    </div>
  );
}

export default AiringTvShow;
