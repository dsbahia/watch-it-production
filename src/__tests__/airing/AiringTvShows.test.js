import React from "react";
import { render, fireEvent } from "@testing-library/react";
import AiringTvShows from "../../components/Upcoming/AiringTvShows";

jest.mock("../../components/MoreDetails", () => {
  return function MockMoreDetails() {
    return <div>MockMoreDetails Component</div>;
  };
});

describe("AiringTvShows", () => {
  it("renders correctly", () => {
    const props = {
      posterpath: "posterpath.jpg",
      title: "Test Tv Show",
      TvId: 1,
    };
    const { asFragment } = render(<AiringTvShows {...props} />);
    expect(asFragment).toMatchSnapshot();
  });

  it("displays tv show title and posterpath correctly", () => {
    const props = {
      posterpath: "posterpath.jpg",
      title: "Test Tv Show",
      TvId: 1,
    };

    const { getByText, getByAltText } = render(<AiringTvShows {...props} />);
    expect(getByText("Test Tv Show")).toBeInTheDocument();
    expect(getByAltText(`${props.title} Tv Show poster`)).toHaveAttribute(
      "src",
      `https://image.tmdb.org/t/p/original/${props.posterpath}`,
    );
  });

  it("toggles More Details on click", () => {
    const props = {
      posterpath: "posterpath.jpg",
      title: "Test Tv Show",
      TvId: 1,
    };
    const { getByText, queryByText } = render(<AiringTvShows {...props} />);
    const moreDetailsButton = getByText("More Details");

    expect(queryByText("MockMoreDetails Component")).toBeNull();

    fireEvent.click(moreDetailsButton);

    expect(queryByText("MockMoreDetails Component")).toBeInTheDocument();
  });
});
