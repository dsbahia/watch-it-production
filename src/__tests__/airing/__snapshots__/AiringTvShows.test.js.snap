// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AiringTvShows renders correctly 1`] = `[Function]`;
